framework/core/physical_device.h

```diff
@@ -158,6 +160,9 @@ class PhysicalDevice
 	// The features that this GPU supports
 	VkPhysicalDeviceFeatures features{};
 
+	// The extensions that this GPU supports
+	std::vector<VkExtensionProperties> device_extensions;
+
```

Comment:
I would make this an `std::unordered_map<std::string, VkExtensionProperties>` so that the code for `is_extension_supported` is a simple `return device_extensions.contains(requested_extension);`

I get this is basically just moved code from `device.h`, but it seems like a little extra work up-front makes the properties easier to work with.

Reply:
I tried this, but I think it requires C++ 20, and the samples project is 17 at the moment?

Reply:
I don't think it requirest C++20 but if it's being problematic, don't let it slow down the actual sample.  

Honestly though I'd love C++20 because it would let us replace lots of functions that take `std::vector` to take a `std::span`, and allow us to pass vectors or `std::array` objects, thus allowing us to declare many current `std::vectors` as `constexpr std::array`.